src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1224,8 +1224,11 @@ def download_historical_logs(resource_group_name, name, log_file=None, slot=None
     logger.warning('Downloaded logs to %s', log_file)
 
 
-def _get_site_credential(client, resource_group_name, name):
-    creds = client.web_apps.list_publishing_credentials(resource_group_name, name)
+def _get_site_credential(client, resource_group_name, name, slot=None):
+    if slot:
+        creds = client.web_apps.list_publishing_credentials_slot(resource_group_name, name, slot)
+    else:
+        creds = client.web_apps.list_publishing_credentials(resource_group_name, name)
```

Comment:
i believe you can simplify to `creds = _generic_site_operation(resource_group_name, name, 'list_publishing_credentials', slot)`